### PR TITLE
Reworks the champion's hardsuit

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

fixes hiero staff recall mesage not working
Reworks the champion's hardsuit
changes:
name: champion's hardsuit -> berserker hardsuit
armor: (MELEE = 65, BULLET = 50, LASER = 50, ENERGY = 60, BOMB = 50, BIO = 100, RAD = 100, FIRE = 80, ACID = 80) -> (MELEE = 30, BULLET = 10, LASER = 10, ENERGY = 20, BOMB = 10, BIO = 100, RAD = 10, FIRE = 100, ACID = 100)
doesnt have ert stripping time so its shorter
it cant hold the pka or cutter on itself, but it carries knives, resonators, spears and the crusher
adds berserk mechanic: with every hit you take thats not from yourself, you get berserk charge equal to the damage dealt to you divided by 4, if its a projectile attack this value is multiplied by 1.5, so a 20 damage melee hit gives you 5 charge, while a 20 damage projectile gives you 7 charge
use berserk ability with 100 charge to go berserk
you move a bit faster, you hit 2 times faster, you get a cool color and your melee armor is massively increased for the duration (about 10 seconds?), but you cannot shoot guns during the effect

showcase video: https://streamable.com/x74xw1

## Why It's Good For The Game

miners getting reskinned ert suits from tendrils which are better than megafauna gear is kinda dumb, this is a more unique melee-based armor which is much more interesting, giving you a choice between reliable upgraded miner suits or the buildup of power with the berserker suit

## Changelog
:cl: Fikou
fix: Fixes hiero staff recall message not working
balance: Champion's hardsuit has been reworked.
spriteadd: fixes spears on suit storages not having icons
/:cl:

### CODE REVIEW

**Comment by IndieanaJones** [comment]
Based

**Comment by Pumpkinoe** [comment]
So do inquisition ERTs just spawn without hardsuits then?

**Comment by Paxilmaniac** [comment]
> So do inquisition ERTs just spawn without hardsuits then?

Inquisition erts get a completely different hardsuit, and this doesn't apply to them in any way

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 1]
```diff
@@ -360,10 +360,6 @@ Contains:
 	actions_types = list()
 	resistance_flags = FIRE_PROOF
 
-/obj/item/clothing/suit/space/hardsuit/ert/paranormal/Initialize()
-	. = ..()
-	AddComponent(/datum/component/anti_magic, FALSE, FALSE, TRUE, ITEM_SLOT_OCLOTHING)
```
Comment (necromanceranne) [inline code/modules/clothing/spacesuits/miscellaneous.dm:365]: I don't see this documented anywhere. The paranornal ERT still use paranormal ert suits even if the bezerker suit was a paranormal suit. Since your bezerker suit is no longer a paranormal hardsuit, why is this change necessary?

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 2]
```diff
@@ -389,19 +385,92 @@ Contains:
...
+		to_chat(owner, "<span class='notice'>Berserk mode is fully charged.</span>")
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/proc/berserk_mode(mob/living/carbon/human/user)
+	to_chat(user, "<span class='warning'>You enter berserk mode.</span>")
+	playsound(user, 'sound/magic/staff_healing.ogg', 50)
+	user.add_movespeed_modifier(/datum/movespeed_modifier/berserk)
+	user.physiology.armor.melee += 50
+	user.next_move_modifier *= 0.5
+	user.add_atom_colour("#950a0a", TEMPORARY_COLOUR_PRIORITY)
+	ADD_TRAIT(user, TRAIT_CHUNKYFINGERS, BERSERK_TRAIT)
```
Comment (necromanceranne) [inline code/modules/clothing/spacesuits/miscellaneous.dm:460]: Add the TRAIT_NOGUNS trait instead. This one is for carbons with big hands, like golems.

File: code/modules/clothing/spacesuits/miscellaneous.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/miscellaneous.dm:365]: the same proc appears later in the code except with arguments for everything being TRUE

File: code/modules/clothing/spacesuits/miscellaneous.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/miscellaneous.dm:460]: will do thanks

**Review by Twaticus — APPROVED** [review]

**Comment by Fikou** [comment]
wait guys this will lower my gbp by 10 can someone change this to balance label HELP!!

**Comment by D4C-420** [comment]
Maybe lower the amount of hit's required to active berserk because lets assume your facing even a plain old goliath, with 30 melee ( or assuming the miner got a sec jumpsuit and hunter charm to raise that to about 50) taking 20 hits is going to get you killed. Needless to say against even lava land regular lava land mobs that many hits no matter how good your healing is just going to get you killed and for all intents and purposes round removed. This also scales like shit when fighting megafauna. It might work better if it scaled with damage taken so you can't just get a miner friend to tickle you with a gas mask and let you store a berserk. Also take out the anti-magic trait. Sure its for fluff (Khorne hates magic and all that) but for balance getting a champion hardsuit should not make you a bigger problem for most cultists and wizards than a Chaplin. 

Overall its a good change but it would relegate the champions hardsuit to another "great I got shit" item to the tendril loot table if the requirements to activate berserk ain't lowered. On a side note maybe buff the base melee to be 40 instead of 30 at the very least.

**Comment by Fikou** [comment]
>because lets assume your facing even a plain old goliath, with 30 melee ( or assuming the miner got a sec jumpsuit and hunter charm to raise that to about 50) taking 20 hits is going to get you killed.

thats not the purpose, you arent supposed to be revved up against every mob encounter, its a bonus you receive while doing stuff and then you can spend it in a challenging situation

>This also scales like shit when fighting megafauna. 

this will probably happen once per megafauna fight and thats fine, the mobs where you wont get damaged enough times are probably colossus and legion, maybe drake

>It might work better if it scaled with damage taken so you can't just get a miner friend to tickle you with a gas mask and let you store a berserk. 

it only works if you take damage i think, but sure i can do this

>Also take out the anti-magic trait. 

khorne hate magic, miner already gets other antimagic items and this was previously antimagic so i dont think its a problem and someone that hates magic should remove it in a separate pr

**Comment by D4C-420** [comment]
> khorne hate magic, miner already gets other antimagic items and this was previously antimagic so i dont think its a problem and someone that hates antimagic should remove it in a separate pr
Fair enough. However the only other anti magic item I can think of is that one spear from the Ravatar ruin which only gets a pass cause its not a piece of armor you have to strip off someone but a big as weapon that does less than a pka and can be removed by conventional means. like shoves, stuns, slips, ect.  Also lore does not matter for gameplay. The bigger reason why champion armor should not have anti-magic is because miners have access to 2 magic spells. Instant recall and sacred flame which you cant use with Champion armor which sucks. At this point it would be better described as closer to the "berserker" armor from Berserk but I'm guessing that's the place you got the idea from. 

> thats not the purpose, you arent supposed to be revved up against every mob encounter, its a bonus you receive while doing stuff and then you can spend it in a challenging situation

Getting hit as a miner is something you avoid doing  especially with only 30 armor. Its not "a bonus you receive while doing stuff" if your punished so hard to get the reward. It punishes players who play well enough to not get hit. Most lava land mobs have simple attack patterns you can kite and megafauna have recognizable attack patterns for this reason but if it scales well enough with damage this is not an issue really. 

Tldr. If it scales decently well with damage then most of my complaints are resolved. The anti-magic on it is dumb and hurts lucky miners but its not the worst issue. The only issues I take with it left are the fact its a worse mining suit if you play well and can probably be power gamed (simple solution just force it's charge to activate when reaching 100. Could still be power gamed but not as badly.) and that it is closer to the Berserk armor from Berserk than Chaos space marine armor but that is again not a big issue.  Good pr.

**Comment by ATH1909** [comment]
I'm fine with the anti-magic, but 20 hits is waaaaaay too many for the intended usage of this hardsuit to be viable. This hardsuit will only be good if you sit still in a corner and let a weak mob (or a buddy) wail on you for a bit to farm up suit charge- in normal gameplay, you're not gonna build up enough charge to pop this thing before you fall into crit unless you spend a shitton of healing meds.

Either reduce the number of hits required to charge up the berserk ability or make it possible to pop your berserk ability while it's partially charged (with the duration of the ability being proportional to how close it was to full charge). Maybe make your berserk meter slowly decrease over time to compensate/make this harder to cheese?

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 3]
```diff
@@ -389,19 +385,94 @@ Contains:
...
+	. = ..()
+	. += "<span class='notice'>Berserk mode is [berserk_charge]% charged.</span>"
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/process(delta_time)
+	. = ..()
+	if(berserk_active)
+		berserk_charge = clamp(berserk_charge -= 10*delta_time, 0, 100)
+	if(!berserk_charge)
+		if(ishuman(loc))
+			end_berserk(loc)
```
Comment (ATH1909) [inline code/modules/clothing/spacesuits/miscellaneous.dm:440]: destroying the suit before the berserk effect ends would let you keep its effects indefinitely

Comment (ATH1909) [reply code/modules/clothing/spacesuits/miscellaneous.dm:440]: actually, I think just _taking the suit off_ before the berserk effect ends would also let you keep its effects indefinitely

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 4]
```diff
@@ -132,6 +132,10 @@
 		var/final_block_chance = wear_neck.block_chance - (clamp((armour_penetration-wear_neck.armour_penetration)/2,0,100)) + block_chance_modifier
 		if(wear_neck.hit_reaction(src, AM, attack_text, final_block_chance, damage, attack_type))
 			return TRUE
+	if(head)
+		var/final_block_chance = head.block_chance - (clamp((armour_penetration-head.armour_penetration)/2,0,100)) + block_chance_modifier
+		if(head.hit_reaction(src, AM, attack_text, final_block_chance, damage, attack_type))
+			return TRUE
```
Comment (ATH1909) [inline code/modules/mob/living/carbon/human/human_defense.dm:138]: 👀

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 5]
```diff
@@ -389,19 +385,94 @@ Contains:
...
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/dropped(mob/user)
+	. = ..()
+	end_berserk(user)
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
```
Comment (ATH1909) [inline code/modules/clothing/spacesuits/miscellaneous.dm:446]: uh, what happens if they go for your chest instead of your head?

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 6]
```diff
@@ -389,19 +385,94 @@ Contains:
...
+	if(berserk_active)
+		return
+	berserk_charge = clamp(berserk_charge += 5, 0, 100)
```
Comment (ATH1909) [inline code/modules/clothing/spacesuits/miscellaneous.dm:449]: would it be possible to multiply the amount of berserk charge gained by 1.5x or something if the thing hitting you is a projectile, since khorne hates those (IIRC)?

File: code/modules/clothing/spacesuits/miscellaneous.dm (hunk 3 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/miscellaneous.dm:440]: no

File: code/modules/clothing/spacesuits/miscellaneous.dm (hunk 5 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/miscellaneous.dm:446]: the same

File: code/modules/clothing/spacesuits/miscellaneous.dm (hunk 6 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/miscellaneous.dm:449]: ill see after i do a test of a miner round with this

File: code/datums/action.dm [hunk 7]
```diff
@@ -304,17 +304,33 @@
...
 	return ..()
 
+/datum/action/item_action/berserk_mode
+	name = "Berserk"
+	desc = "Increase your movement and melee speed while also increasing your melee armor for a short amount of time."
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+	button_icon_state = "berserk_mode"
+	background_icon_state = "bg_demon"
+
+/datum/action/item_action/berserk_mode/Trigger()
```
Comment (ATH1909) [inline code/datums/action.dm:324]: can you double click this button before the first process() with berserk mode on gets run (which is what reduces your berserk_charge to below 100), effectively doubling the benefits of berserk mode?

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 8]
```diff
@@ -389,19 +385,94 @@ Contains:
...
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/proc/end_berserk(mob/living/carbon/human/user)
+	if(!berserk_active)
+		return
+	to_chat(user, "<span class='warning'>You exit berserk mode.</span>")
+	playsound(user, 'sound/magic/summonitems_generic.ogg', 50)
+	user.remove_movespeed_modifier(/datum/movespeed_modifier/berserk)
+	user.physiology.armor.melee -= 50
+	user.next_move_modifier *= 2
+	user.remove_atom_colour(TEMPORARY_COLOUR_PRIORITY, "#950a0a")
+	REMOVE_TRAIT(user, TRAIT_CHUNKYFINGERS, BERSERK_TRAIT)
```
Comment (ATH1909) [inline code/modules/clothing/spacesuits/miscellaneous.dm:474]: berserk mode adds TRAIT_NOGUNS but removes TRAIT_CHUNKYFINGERS

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 9]
```diff
@@ -389,19 +385,94 @@ Contains:
...
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/process(delta_time)
+	. = ..()
+	if(berserk_active)
+		berserk_charge = clamp(berserk_charge -= 10*delta_time, 0, 100)
+	if(!berserk_charge)
+		if(ishuman(loc))
+			end_berserk(loc)
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/dropped(mob/user)
```
Comment (ATH1909) [inline code/modules/clothing/spacesuits/miscellaneous.dm:442]: does this get called if the helmet gets destroyed?

File: code/modules/clothing/spacesuits/miscellaneous.dm (hunk 8 above)
Comment (Fikou) [reply code/modules/clothing/spacesuits/miscellaneous.dm:474]: ?

File: code/datums/action.dm (hunk 7 above)
Comment (Fikou) [reply code/datums/action.dm:324]: i will test

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 10]
```diff
@@ -389,19 +385,104 @@ Contains:
...
+/obj/item/clothing/suit/space/hardsuit/berserker/RemoveHelmet()
+	var/obj/item/clothing/head/helmet/space/hardsuit/berserker/helm = helmet
+	if(helm?.berserk_active)
+		return
+	return ..()
+
+#define MAX_BERSERK_CHARGE 100
+#define PROJECTILE_HIT_MULTIPLIER 1.5
+#define DAMAGE_TO_CHARGE_SCALE 0.25
+#define CHARGE_DRAINED_PER_SECOND 5
```
Comment (Jared-Fogle) [inline code/modules/clothing/spacesuits/miscellaneous.dm:414]: Undef all these at the end.

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 11]
```diff
@@ -389,19 +385,104 @@ Contains:
...
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/berserker
+	armor = list(MELEE = 30, BULLET = 10, LASER = 10, ENERGY = 20, BOMB = 10, BIO = 100, RAD = 10, FIRE = 100, ACID = 100)
+	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
+	resistance_flags = FIRE_PROOF
+	allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/pickaxe, /obj/item/spear, /obj/item/organ/regenerative_core/legion, /obj/item/kitchen/knife, /obj/item/kinetic_crusher, /obj/item/resonator)
+
+
+/obj/item/clothing/suit/space/hardsuit/berserker/Initialize()
+	. = ..()
+	AddComponent(/datum/component/anti_magic, TRUE, TRUE, TRUE, ITEM_SLOT_OCLOTHING)
```
Comment (Jared-Fogle) [inline code/modules/clothing/spacesuits/miscellaneous.dm:403]: Normally I'd ask for these to be named parameters, but the names for these are underscore-prefixed, so consider this a note to myself to fix that.

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 12]
```diff
@@ -389,19 +385,104 @@ Contains:
...
+	ADD_TRAIT(src, TRAIT_NODROP, LOCKED_HELMET_TRAIT)
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/examine()
+	. = ..()
+	. += "<span class='notice'>Berserk mode is [berserk_charge]% charged.</span>"
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/process(delta_time)
+	. = ..()
+	if(berserk_active)
+		berserk_charge = clamp(berserk_charge -= CHARGE_DRAINED_PER_SECOND * delta_time, 0, MAX_BERSERK_CHARGE)
```
Comment (Jared-Fogle) [inline code/modules/clothing/spacesuits/miscellaneous.dm:442]:
```suggestion
		berserk_charge = clamp(berserk_charge - CHARGE_DRAINED_PER_SECOND * delta_time, 0, MAX_BERSERK_CHARGE)
```

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 13]
```diff
@@ -389,19 +385,104 @@ Contains:
...
+	. = ..()
+	end_berserk(user)
+
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(berserk_active)
+		return
+	var/berserk_value = damage * DAMAGE_TO_CHARGE_SCALE
+	if(attack_type == PROJECTILE_ATTACK)
+		berserk_value *= PROJECTILE_HIT_MULTIPLIER
+	berserk_charge = clamp(round(berserk_charge += berserk_value, 1), 0, MAX_BERSERK_CHARGE)
```
Comment (Jared-Fogle) [inline code/modules/clothing/spacesuits/miscellaneous.dm:457]:
```suggestion
	berserk_charge = clamp(round(berserk_charge + berserk_value, 1), 0, MAX_BERSERK_CHARGE)
```

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 14]
```diff
@@ -389,19 +385,104 @@ Contains:
...
+	if(berserk_charge >= MAX_BERSERK_CHARGE)
+		to_chat(owner, "<span class='notice'>Berserk mode is fully charged.</span>")
+
+/// Starts berserk, giving the wearer 50 melee armor, doubled attacking speed, NOGUNS trait, adding a color and giving them the berserk movespeed modifier
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/proc/berserk_mode(mob/living/carbon/human/user)
+	to_chat(user, "<span class='warning'>You enter berserk mode.</span>")
+	playsound(user, 'sound/magic/staff_healing.ogg', 50)
+	user.add_movespeed_modifier(/datum/movespeed_modifier/berserk)
+	user.physiology.armor.melee += 50
+	user.next_move_modifier *= 0.25
```
Comment (Jared-Fogle) [inline code/modules/clothing/spacesuits/miscellaneous.dm:467]: These should ideally be defines as well.

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 15]
```diff
@@ -389,19 +385,104 @@ Contains:
...
+	user.add_atom_colour("#950a0a", TEMPORARY_COLOUR_PRIORITY)
```
Comment (Jared-Fogle) [inline code/modules/clothing/spacesuits/miscellaneous.dm:468]: Move the color into a define in `code/__DEFINES/colors.dm`.

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 16]
```diff
@@ -389,19 +385,104 @@ Contains:
...
+
+/// Ends berserk, reverting the changes from the proc [berserk_mode]
+/obj/item/clothing/head/helmet/space/hardsuit/berserker/proc/end_berserk(mob/living/carbon/human/user)
+	if(!berserk_active)
+		return
+	to_chat(user, "<span class='warning'>You exit berserk mode.</span>")
+	playsound(user, 'sound/magic/summonitems_generic.ogg', 50)
+	user.remove_movespeed_modifier(/datum/movespeed_modifier/berserk)
+	user.physiology.armor.melee -= 50
+	user.next_move_modifier *= 4
```
Comment (Jared-Fogle) [inline code/modules/clothing/spacesuits/miscellaneous.dm:481]: Prime example of why this should be a define, any balance changes to this would have to be duplicated over here as well. 😛

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 17]
```diff
@@ -389,19 +385,104 @@ Contains:
...
+	user.remove_atom_colour(TEMPORARY_COLOUR_PRIORITY, "#950a0a")
```
Comment (Jared-Fogle) [inline code/modules/clothing/spacesuits/miscellaneous.dm:482]: `950a0a` should be in the colors defines like I said earlier, though since its being duplicated you might want another define named `BERSERK_COLOR` or similar, again if for no other reason than to lower code needing to be changed.

**Comment by Taylorkills2** [comment]
Depends on how you can accumulate hits. 
Looks like this will really only get good use by using the Katana or if you're an Ashligger. 
Though watching an Ashligger using this and a Katana on station will be hilarious to watch them go mow through security. like a retardedly fast lawn mower. 
I quiver in excitement.

**Comment by Ravellon** [comment]
Melee 30.
Explorer's suit with 3 goliath plates is 60.
This is literally worse than explorer's suit.
